internal/headers.go

```diff
@@ -45,9 +46,20 @@ func AddTrailers(
 	}
 }
 
-// ConvertToProtoHeader converts HTTP headers to a slice of proto Headers.
-func ConvertToProtoHeader(
-	src http.Header,
+// ConvertHTTPHeaderToProtoHeader converts HTTP headers to a slice of proto Headers.
+func ConvertHTTPHeaderToProtoHeader(src http.Header) []*v1.Header {
+	return convertToProtoHeader(src)
+}
+
+// ConvertQueryParamToProtoHeader converts query parameters to a slice of proto Headers.
+func ConvertQueryParamToProtoHeader(
+	src url.Values,
+) []*v1.Header {
+	return convertToProtoHeader(src)
+}
```

Comment:
Not sure I see the value of these two specialized methods. Why not just change the type of the one, exported `ConvertToProtoHeader` to accept `map[string][]string` instead of `http.Header` and then document in the comments that it can accept either `http.Header` or `url.Values`.

Reply:
Ah ok. I figured we'd want the type safety of passing either `url.Value` or `http.Header`, but using the map as the public API is cool with me.